src/deferred.rs

```diff
@@ -19,55 +20,60 @@ pub struct Deferred {
     data: Data,
 }
 
+/// `Deferred::new` requires that another thread may execute the inner function.
+unsafe impl Send for Deferred {}
+
+impl fmt::Debug for Deferred {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        write!(f, "Deferred {{ ... }}")
+    }
+}
+
 impl Deferred {
     /// Constructs a new `Deferred` from a `FnOnce()`.
-    pub fn new<F: FnOnce()>(f: F) -> Self {
+    ///
+    /// # Safety
+    ///
+    /// It should be safe for another thread to execute the given function.
+    pub unsafe fn new<F: FnOnce()>(f: F) -> Self {
```

Comment:
Is there a `Send`/`Sync` for functions? Does `F: FnOnce() + Send + Sync` work, or is this already the case?

Reply:
In this case we only need `F: FnOnce() + Send`.

Reply:
@Amanieu I think you're right `Sync` isn't needed.